pyanaconda/modules/storage/partitioning/interactive_utils.py

```diff
@@ -284,19 +286,151 @@ def validate_label(label, fmt):
 
     :param str label: a label
     :param DeviceFormat fmt: a device format to label
-    :return: a list of error messages
+    :return: an error message
     """
-    errors = []
+    if not label:
+        return None
 
     if fmt.exists:
-        errors.append(_("Cannot relabel already existing file system."))
-    elif not fmt.labeling():
-        if label != "":
-            errors.append(_("Cannot set label on file system."))
-    elif not fmt.label_format_ok(label):
-        return errors.append(_("Unacceptable label format for file system."))
+        return _("Cannot relabel already existing file system.")
+
+    if not fmt.labeling():
+        return _("Cannot set label on file system.")
+
+    if not fmt.label_format_ok(label):
+        return _("Unacceptable label format for file system.")
+
+    return None
+
+
+def validate_mount_point(path, mount_points):
+    """Validate the given path of a mount point.
+
+    :param path: a path to validate
+    :param mount_points: a list of existing mount points
+    :return: an error message
+    """
+    system_mount_points = ["/dev", "/proc", "/run", "/sys"]
+
+    if path in mount_points:
+        return _("That mount point is already in use. Try something else?")
+
+    if not path:
+        return _("Please enter a valid mount point.")
+
+    if path in system_mount_points:
+        return _("That mount point is invalid. Try something else?")
+
+    if ((len(path) > 1 and path.endswith("/")) or
```

Comment:
This looks like something that could be in a separate method & could be unit tested that way. :)

Reply:
I am afraid to touch this code at this moment. I will revisit it later.

Reply:
It would be great to give user better understanding of what is wrong. In other words have a message for every part of the check. However, yes, this could be done later.